File: backend/models/user_profile.py
from mongoengine import Document, StringField, ListField, connect

connect('game_analysis')

class UserProfile(Document):
    username = StringField(required=True, unique=True)
    email = StringField(required=True, unique=True)
    preferences = ListField(StringField())

if __name__ == '__main__':
    user = UserProfile(username='player1', email='player1@example.com', preferences=['aggressive', 'sniper'])
    user.save()

File: backend/models/UserProfile.js
const mongoose = require('mongoose');

// User Profile Schema definition
const userProfileSchema = new mongoose.Schema({
    username: {
        type: String,
        required: true,
        unique: true,
        trim: true
    },
    email: {
        type: String,
        required: true,
        unique: true,
        lowercase: true,
        trim: true
    },
    password: {
        type: String,
        required: true,
        minlength: 6
    },
    createdAt: {
        type: Date,
        default: Date.now
    },
    updatedAt: {
        type: Date,
        default: Date.now
    },
    preferences: {
        type: Object,
        default: {}
    },
    gameplayRecords: [{
        type: mongoose.Schema.Types.ObjectId,
        ref: 'GameplayRecord'
    }]
}, { timestamps: true });

// Export the UserProfile model
module.exports = mongoose.model('UserProfile', userProfileSchema);